concurrent/einstein_test.go

```diff
@@ -0,0 +1,250 @@
+		return micro.DisjointO(
+			leftOf(x, y),
+			rightOf(x, y),
+		)
+	}
+
+	anon := func() *ast.SExpr {
+		return ast.NewVariable(time.Now().String())
+	}
+
+	streetDef := ast.NewList(
+		ast.NewList(one, anon(), anon(), anon(), anon(), anon()),
+		ast.NewList(two, anon(), anon(), anon(), anon(), anon()),
+		ast.NewList(three, anon(), anon(), anon(), anon(), anon()),
+		ast.NewList(four, anon(), anon(), anon(), anon(), anon()),
+		ast.NewList(five, anon(), anon(), anon(), anon(), anon()),
+	)
+
+	streetEq := func(q *ast.SExpr) micro.Goal {
+		return micro.EqualO(
+			q,
+			streetDef,
+		)
+	}
+
+	mUnrolled := memberOUnrolled(streetDef)
+
+	solution := func(street, fishowner, a, b, c, d, e, f, g, h, i, j *ast.SExpr) micro.Goal {
+		return mini.ConjPlus(
+			streetEq(street),
+			mUnrolled(ast.NewList(anon(), ast.NewSymbol("brit"), ast.NewSymbol("red"), anon(), anon(), anon())),
+			mUnrolled(ast.NewList(anon(), ast.NewSymbol("swede"), anon(), ast.NewSymbol("dog"), anon(), anon())),
+			mUnrolled(ast.NewList(anon(), ast.NewSymbol("dane"), anon(), anon(), ast.NewSymbol("tea"), anon())),
+			mUnrolled(ast.NewList(a, anon(), ast.NewSymbol("green"), anon(), anon(), anon())),
+			mUnrolled(ast.NewList(b, anon(), ast.NewSymbol("white"), anon(), anon(), anon())),
+			leftOf(a, b),
+			mUnrolled(ast.NewList(anon(), anon(), ast.NewSymbol("green"), anon(), ast.NewSymbol("coffee"), anon())),
+			mUnrolled(ast.NewList(anon(), anon(), anon(), ast.NewSymbol("birds"), anon(), ast.NewSymbol("pall-mall"))),
+			mUnrolled(ast.NewList(anon(), anon(), ast.NewSymbol("yellow"), anon(), anon(), ast.NewSymbol("dunhill"))),
+			mUnrolled(ast.NewList(three, anon(), anon(), anon(), ast.NewSymbol("milk"), anon())),
+			mUnrolled(ast.NewList(one, ast.NewSymbol("norwegian"), anon(), anon(), anon(), anon())),
+			mUnrolled(ast.NewList(c, anon(), anon(), anon(), anon(), ast.NewSymbol("blend"))),
+			mUnrolled(ast.NewList(d, anon(), anon(), ast.NewSymbol("cats"), anon(), anon())),
+			nextTo(c, d),
+			mUnrolled(ast.NewList(e, anon(), anon(), ast.NewSymbol("horse"), anon(), anon())),
+			mUnrolled(ast.NewList(f, anon(), anon(), anon(), anon(), ast.NewSymbol("dunhill"))),
+			nextTo(e, f),
+			mUnrolled(ast.NewList(anon(), anon(), anon(), anon(), ast.NewSymbol("beer"), ast.NewSymbol("bluemaster"))),
+			mUnrolled(ast.NewList(anon(), ast.NewSymbol("german"), anon(), anon(), anon(), ast.NewSymbol("prince"))),
+			mUnrolled(ast.NewList(g, ast.NewSymbol("norwegian"), anon(), anon(), anon(), anon())),
+			mUnrolled(ast.NewList(h, anon(), ast.NewSymbol("blue"), anon(), anon(), anon())),
+			nextTo(g, h),
+			mUnrolled(ast.NewList(i, anon(), anon(), anon(), anon(), ast.NewSymbol("blend"))),
+			mUnrolled(ast.NewList(j, anon(), anon(), anon(), ast.NewSymbol("water"), anon())),
+			nextTo(i, j),
+			mUnrolled(ast.NewList(anon(), fishowner, anon(), ast.NewSymbol("fish"), anon(), anon())),
+		)
+	}
+
+	sexprs := micro.Run(-1, func(q *ast.SExpr) micro.Goal {
```

Comment:
Maybe it is possible to separate the creation and running, so that creation of all the objects happen before the benchmark starts and then only do micro.Run inside the benchmark.

I think since it is so many objects that we need to create, it might be contributing some noise.